src/overlays/actors/ovl_En_Heishi2/z_en_heishi2.c

```diff
@@ -21,77 +69,852 @@ const ActorInit En_Heishi2_InitVars = {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005C30, 1.0f, 0.0f, (s16)frameCount, 0, -10.0f);
+    this->actionFunc = func_80A5399C;
+}
+
+void func_80A5399C(EnHeishi2* this, GlobalContext* globalCtx) {
+    s16 var;
+
+    this->unk_30B = 0;
+    var = 0;
+    if ((gSaveContext.infTable[7] & 0x40) != 0) {
+        if ((gSaveContext.infTable[7] & 0x80) == 0) {
+            if (func_8008F080(globalCtx) == 1) {
+                if (this->unk_309 == 0) {
+                    this->actor.textId = 0x200A; // "Wha-ha-ha-hah! Do you think you're in disguise, Mr. Hero?"
+                } else {
+                    this->actor.textId = 0x200B; // "Will you sell it to me? Yes/No way"
+                }
+                this->unk_300 = 4;
+                this->unk_30B = 1;
+                var = 1;
+            } else {
+                this->actor.textId =
+                    0x2016; // "I wish I could go to the mask shop in town to buy a present for my kid..sigh.."
+                this->unk_300 = 6;
+                var = 1;
+            }
+        } else {
+            this->actor.textId = 0x2020; // "My boy will be very happy with this!"
+            this->unk_300 = 5;
+            this->unk_30E = 0;
+        }
+        if (Text_GetFaceReaction(globalCtx, 5) != 0) {
+            if (var == 0) {
+                this->actor.textId = Text_GetFaceReaction(globalCtx, 5);
+                this->unk_30B = 1;
+                this->unk_300 = 6;
+                this->unk_30E = 0;
+            }
+        }
+        this->actionFunc = func_80A5475C;
+    } else {
+        // "I don't know"
+        osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆☆☆☆ とおしゃしねぇちゅーの ☆☆☆☆☆ \n" VT_RST);
+        this->actionFunc = func_80A53AD4;
+    }
+}
+
+void func_80A53AD4(EnHeishi2* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 flag;
+    s16 angleDiff;
+    s16 yawDiff;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (Text_GetFaceReaction(globalCtx, 5) != 0) {
+        this->actor.textId = Text_GetFaceReaction(globalCtx, 5);
+    } else {
+        this->actor.textId = 0x200E; // "The road is closed beyond this point! Can't you read..."
+    }
+    this->unk_300 = 6;
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        flag = func_8002F368(globalCtx);
+        if (flag == 1) {
+            func_80078884(NA_SE_SY_CORRECT_CHIME);
+            player->actor.textId = 0x2010; // "Oh, this is...this is surely Princess Zelda's handwriting!"
+            this->unk_300 = 5;
+            this->actionFunc = func_80A53C0C;
+        } else if (flag != 0) {
+            player->actor.textId = 0x200F; // "I don't want that!"
+        }
+    } else {
+        angleDiff = this->actor.rotTowardsLinkY - this->actor.shape.rot.y;
+        if (angleDiff >= 0) {
+            yawDiff = angleDiff;
+        } else {
+            yawDiff = -angleDiff;
+        }
+        if (!(120.0f < this->actor.xzDistanceFromLink)) {
+            if (yawDiff < 0x4300) {
```

Comment:
saw you changed the name, but you still need to use the ABS() macro. That will also get rid of the need of two temps
```suggestion
        yawDiff = this->actor.rotTowardsLinkY - this->actor.shape.rot.y;
        if (!(120.0f < this->actor.xzDistanceFromLink) && (ABS(yawDiff) < 0x4300) {
```

Reply:
Only matched with the temp.